pkg/ddevapp/craftcms.go

```diff
@@ -0,0 +1,170 @@
+package ddevapp
+
+import (
+	"fmt"
+	"github.com/drud/ddev/pkg/archive"
+	"github.com/drud/ddev/pkg/fileutil"
+	"github.com/drud/ddev/pkg/util"
+	"github.com/pkg/errors"
+	"os"
+	"path/filepath"
+	"regexp"
+)
+
+// isCraftCmsApp returns true if the app is of type craftcms
+func isCraftCmsApp(app *DdevApp) bool {
+	return fileutil.FileExists(filepath.Join(app.AppRoot, "craft"))
```

Comment:
This seems slightly suspicious as a fundamental detector, but you're the Craft expert(s). Completely valid? Any idea why the craft script is placed in the app root instead of some place like bin or vendor/bin ?

Reply:
That's where it goes, it is the `cli` for Craft. It's at the root, I presume, to make it convenient to run from the project root